pygmt/datasets/tile_map.py

```diff
@@ -16,7 +17,15 @@
 __doctest_requires__ = {("load_tile_map"): ["contextily"]}
 
 
-def load_tile_map(region, zoom="auto", source=None, lonlat=True, wait=0, max_retries=2):
+def load_tile_map(
+    region,
+    zoom="auto",
+    source=None,
+    lonlat=True,
+    wait=0,
+    max_retries=2,
+    zoom_adjust=None,
```

Comment:
Add type hints to this newly added parameter?
```suggestion
    zoom_adjust: int | None,
```

Reply:
Done at f0c95d89c77506993e42dc0e59bcf21eedcb9bae for just the `zoom_adjust` parameter. Since all the lines are changing, should I just add type hints for all the other parameters too?

Reply:
Perfer to focus on the `zoom_adjust` parameter only in this PR.